core/src/main/java/jenkins/model/lazy/LazyBuildMixIn.java

```diff
@@ -105,7 +105,9 @@ public void onLoad(ItemGroup<? extends Item> parent, String name) throws IOExcep
         int max = _builds.maxNumberOnDisk();
         int next = asJob().getNextBuildNumber();
         if (next <= max) {
-            LOGGER.log(Level.WARNING, "JENKINS-27530: improper nextBuildNumber {0} detected in {1} with highest build number {2}; adjusting", new Object[] {next, asJob(), max});
+            if (next != 0 && max != 0) {
```

Comment:
is this not something that should be fixed in the matrix plugin to set this properly?

Reply:
AFAICT, just setting return 0; to return 1; should solve the issue
https://github.com/jenkinsci/matrix-project-plugin/blob/94e4f49a252895bc010c84afe81cc858c285d760/src/main/java/hudson/matrix/MatrixConfiguration.java#L206